packages/core/src/routes/interaction/mfa.ts

```diff
@@ -171,7 +171,7 @@ export default function mfaRoutes<T extends IRouterParamContext>(
       } = ctx;
 
       assertThat(
-        policy === MfaPolicy.UserControlled,
+        policy === MfaPolicy.PromptAtSignInAndSignUp || policy === MfaPolicy.PromptOnlyAtSignIn,
```

Comment:
How about just `policy !== MfaPolicy.mandatory`.  `MfaPolicy.NoPrompt` does not hurt at this point. Keep the logic simple.